main.js

```diff
@@ -0,0 +1,331 @@
+
+  post.appendChild(newCommentForm);
+}
+
+// Update post display
+function displayPosts (currentPosts) {
+  postsDiv.innerHTML = '';
+  
+  currentPosts.forEach (function (post) {
+    const postDiv = document.createElement('div');
+    postDiv.setAttribute('data-index', post.index);
+    postDiv.setAttribute('class', 'new-post');
+
+    const messageDiv = document.createElement('div')
+    messageDiv.innerHTML = post.message;
+
+    const postedByDiv = document.createElement('div')
+    postedByDiv.innerHTML = `Posted By: <strong>${post.name}</strong>`;
+
+    const iconDiv = document.createElement('div');
+    iconDiv.setAttribute('class', 'icons');
+    iconDiv.innerHTML = '<i class="fa-solid fa-comment"></i> <i class="fa-solid fa-pen-to-square post-icon"></i> <i class="fa-solid fa-trash post-icon"></i> <hr>';
+
+    const commentsDiv = document.createElement('div');
+    commentsDiv.setAttribute('class', 'comments col-md-12 offset-md-12');
+
+    postDiv.appendChild(messageDiv);
+    postDiv.appendChild(postedByDiv);
+    postDiv.appendChild(iconDiv);
+
+    displayEditPostForm(postDiv);
+    
+    postDiv.appendChild(commentsDiv);
+
+    displayNewCommentForm(postDiv);
+
+    postsDiv.appendChild(postDiv);
+
+    messageInput.value = '';
+    nameInput.value = '';
+
+    addSubmitCommentEventListeners();
+    addToggleCommentsEventListeners();
+    addToggleEditPostEventListeners();
+    addDeletePostEventListeners();
+    addEditPostEventListeners();
+  })
+}
+
+
+//
+// Comment functions
+//
+
+// Return a new comment object
+function createNewComment (name, message) {
+    return Comment(name, message);
+}
+
+// Create new comment object and add to comment array in associated post, update display
+function addComment (e) {
+  const post = getPost(e);
+  const postIndex = post.dataset.index;
+
+  const commentNameInput = post.querySelector('#comment-name');
+  const commentMessageInput = post.querySelector('#comment-message');
+
+  const newComment = createNewComment(commentNameInput.value, commentMessageInput.value);
+
+  commentNameInput.value = '';
+  commentMessageInput.value = '';
+
+  myPosts.posts[postIndex].addCommentToPost(newComment);
+
+  displayComments(post, myPosts.posts[postIndex].comments);
+}
+
+// Delete comment from comment array in associated post, update display
+function deleteComment (e) {
+  const post = getPost(e);
+  const postIndex = post.dataset.index;
+
+  const comment = getComment(e);
+  const commentIndex = comment.dataset.index;
+
+  myPosts.posts[postIndex].comments.splice(commentIndex, 1);
+
+  myPosts.posts[postIndex].comments.forEach((comment) => comment.index =  myPosts.posts[postIndex].comments.indexOf(comment));
```

Comment:
Would be easier to read if you changed to

myPosts.posts[postIndex].comments.forEach((comment) => {
    comment.index = myPosts.posts[postIndex].comments.indexOf(comment);
});